File: src/test/java/coreTDD/rectangleIntersection/IntersectionDetectorTest.java
package coreTDD.rectangleIntersection;

import org.junit.Test;
import coreTDD.rectangleIntersection.helpfulClasses.CoordinateRectangle;
import coreTDD.rectangleIntersection.helpfulClasses.Coordinates;

import static org.assertj.core.api.Assertions.assertThat;

public class IntersectionDetectorTest {

    private IntersectionDetector intersectionDetector = new IntersectionDetector();

    @Test
    public void non_overlapping_rectangles_do_not_intersect() {
        assertThat(intersectionDetector.rectangleIntersection(
                new CoordinateRectangle(
                        new Coordinates(0, 0),
                        new Coordinates(1, 1)
                ),
                new CoordinateRectangle(
                        new Coordinates(2, 2),
                        new Coordinates(3, 3)
                ))
        ).isFalse();
    }
}

File: src/test/kotlin/kotlinCoreTDDTests/rectangleIntersection/IntersectionDetectorTest.kt
package kotlinCoreTDDTests.rectangleIntersection

import kotlinCoreTDD.rectangleIntersection.IntersectionDetector
import kotlinCoreTDD.rectangleIntersection.helpfulClasses.Coordinate
import kotlinCoreTDD.rectangleIntersection.helpfulClasses.CoordinateRectangle
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class IntersectionDetectorTest {

    @Test
    fun rectangles__ZeroZero_OneOne__TwoTwo_ThreeThree__doNotIntersect() {
        val rectangleIntersection = IntersectionDetector().rectangleIntersection(
                CoordinateRectangle(
                        Coordinate(x = 0, y = 0),
                        Coordinate(x = 1, y = 1)
                ),
                CoordinateRectangle(
                        Coordinate(x = 2, y = 2),
                        Coordinate(x = 3, y = 3)
                )
        )

        assertThat(rectangleIntersection).isFalse()
    }
}